### ISSUE TITLE
TestMeshNetworking flake from deadlock

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#29341]
https://prow.istio.io/view/gs/istio-prow/logs/unit-tests_istio_postsubmit/1334134193264791552

Test timesout

Blocked on
```
func (s *DiscoveryServer) ConfigUpdate(req *model.PushRequest) {
	inboundConfigUpdates.Increment()
	s.pushChannel <- req
}
```

pushChannel has capacity 10. So something is not reading from it

There are also 6 discovery servers running - either the test is doing something wrong or we are not shutting them down after some tests?

I don't see any `handleUpdates` goroutine running..

**Comment by howardjohn** [comment]
I can consistently reproduce by setting pushChannel capacity to 1

**Comment by stevenctl** [comment]
We start filling the update queue while initializing k8s controller (during the force-sync)

`DiscoveryServer.Start` is where we run `handleUpdates`.. trying to look at the differences between our actual bootstrap and the fake to understand why this deadlock doesn't happen in a real pilot instance.

**Comment by howardjohn** [comment]
Its plausible it does happen but we are doing things in parallel? Just a guess

**Comment by stevenctl** [comment]
We start the handleUpdates  and sendPushes here (as a startFunc):

https://github.com/istio/istio/blob/3c5c4885532a917d3da645565875becafb998369/pilot/pkg/bootstrap/server.go#L350-L353

before actually waiting for synced caches

https://github.com/istio/istio/blob/3c5c4885532a917d3da645565875becafb998369/pilot/pkg/bootstrap/server.go#L383-L388

Allowing readiness probes/gRPC server to start is protected by cache sync though, so we won't actually send any pushes until caches are synced. I don't think we'd have real issues in real pilot, but it could be good to move the `handleUpdates` start up somewhere earlier.

**Comment by stevenctl** [comment]
should be fixed by #29346 but I'll keep an eye out in CI and locally for failures.
